src/modding/ModManager.cs

```diff
@@ -408,6 +632,92 @@ public override void _Process(float delta)
         wasVisible = isCurrentlyVisible;
     }
 
+    public void UpdateLoadPosition(int startIndex = 0)
+    {
+        if (enabledMods != null)
+        {
+            for (int index = startIndex; index < enabledMods.Count; ++index)
+            {
+                enabledMods[index].LoadPosition = index;
+            }
+        }
+    }
+
+    /// <summary>
+    ///   Gets a array of FullModDetails of all the mods that have a config file
+    /// </summary>
+    public List<FullModDetails> GetAllConfigurableMods()
+    {
+        List<FullModDetails> resultArray = new List<FullModDetails>();
+        if (enabledMods != null)
+        {
+            foreach (FullModDetails currentMod in enabledMods)
+            {
+                if (currentMod.CurrentConfiguration != null)
+                {
+                    resultArray.Add(currentMod);
+                }
+            }
+        }
+
+        return resultArray;
+    }
+
+    /// <summary>
+    ///   This saves the Mod Settings/Config to a file
+    /// </summary>
+    /// <returns> True on success, false if the file can't be written .</returns>
+    /// <remarks>
+    ///   This was based on the Save method from Settings.cs
+    /// </remarks>
+    public bool SaveAllModsSettings()
+    {
+        using var file = new File();
+        var error = file.Open(Constants.MOD_CONFIGURATION_FILE, File.ModeFlags.Write);
+
+        if (error != Error.Ok)
+        {
+            GD.PrintErr("Couldn't open mod configuration file for writing.");
+            return false;
+        }
+
+        var modConfigList = GetAllConfigurableMods();
+        Dictionary<string, Dictionary<string, object>> savedConfig =
+            new Dictionary<string, Dictionary<string, object>>();
+        foreach (FullModDetails currentMod in modConfigList)
+        {
+            savedConfig.Add(currentMod.InternalName,
+                currentMod.CurrentConfiguration ?? new Dictionary<string, object>());
+        }
+
+        file.StoreString(JsonConvert.SerializeObject(savedConfig, Formatting.Indented));
+        file.Close();
+
+        return true;
+    }
+
+    /// <summary>
+    ///   This loads the all of the Mod Settings/Config from a file
+    /// </summary>
+    /// <returns> The SavedConfig on success, null if the file can't be read.</returns>
+    public Dictionary<string, Dictionary<string, object>> LoadAllModsSettings()
+    {
+        using var file = new File();
+        var error = file.Open(Constants.MOD_CONFIGURATION_FILE, File.ModeFlags.Read);
+
+        if (error != Error.Ok)
+        {
+            GD.PrintErr("Couldn't open mod configuration file for reading.");
+            return null!;
+        }
```

Comment:
Or maybe it would be better to not create the file, but not fail if it doesn't exist. It seems not everyone will run mods so having an extra file be created for no reason doesn't seem like the best option.

Reply:
I made it not fail when run by just checking if it null